### PR TITLE
Implement GPIO for nuttx

### PR DESCRIPTION
**Pull request by hs0225**
Add GPIO API for nuttx https://github.com/Samsung/iotjs/issues/128
- initialize, release
- setPin, writePin, readPin
- setPort, writePort, readPort

Fix test app

IoT.js-DCO-1.0-Signed-off-by: Hosung Kim hs852.kim@samsung.com

### CODE REVIEW

File: src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx.cpp [hunk 1]
```diff
@@ -48,6 +126,8 @@ Gpio* Gpio::Create(JObject& jgpio) {
 
 GpioArmNuttx::GpioArmNuttx(JObject& jgpio)
     : Gpio(jgpio) {
+  _fd = NULL;
+  _initialized = false;
```
Comment (zherczeg) [inline src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx.cpp]: you should initialize these like Gpio above:

    : Gpio(jgpio)
    , _fd(NULL)
    ...

File: src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx.cpp [hunk 2]
```diff
@@ -57,51 +137,311 @@ GpioArmNuttx* GpioArmNuttx::GetInstance()
...
+                 gpioPort[port_index] | gpioPin[pin_index + offset];
+
+    if (gpio_ioctl_config(&cdata)) {
+      req_data->result = kGpioErrSys;
+      return;
+    }
+  }
+  req_data->result = kGpioErrOk;
+
+}
```
Comment (zherczeg) [inline src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx.cpp]: I think these extra lines are unnecessary.

**Comment by zherczeg** [comment]
LGTM after style fixes

File: src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx.cpp (hunk 1 above)
Comment (hs0225) [inline src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx.cpp]: updated as commented

**Pull request merged**